### PR TITLE
Adding linting and unit tests

### PR DESCRIPTION
**Pull request by filmaj**
###  Summary

Adding first pass linting and basic unit tests.

### CODE REVIEW

**Comment by filmaj** [comment]
FYI I have not integrated this into the GitHub workflow file as I believe @stevengill is working on revamping that. So, after that lands, we can integrate running the tests into the CI by making sure the workflow runs `npm test`.

File: src/slack-send.js [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+    let payload = core.getInput('payload');
+
+    if (payload) {
+      try {
+        // confirm it is valid json
+        payload = JSON.parse(payload);
+      } catch (e) {
+        // passed in payload wasn't valid json
+        console.error('passed in payload was invalid JSON');
```
Comment (seratch) [inline src/slack-send.js:35]: I think it's fine to use `console` for this purpose. Should we ignore "no-console" warnings for this entire package?

```
> slack-github-action@1.15.0 lint
> eslint .


/Users/ksera/github/slack-github-action/src/slack-send.js
  35:9  warning  Unexpected console statement  no-console
  49:9  warning  Unexpected console statement  no-console
  56:9  warning  Unexpected console statement  no-console
  77:9  warning  Unexpected console statement  no-console
  78:9  warning  Unexpected console statement  no-console

✖ 5 problems (0 errors, 5 warnings)
```

**Review by seratch — APPROVED** [review]
Great improvement! Left a comment about the eslint warnings

File: src/slack-send.js (hunk 1 above)
Comment (filmaj) [reply src/slack-send.js:35]: Yes that makes sense. I'll add it to the eslint rules for this repo.

**Review by stevengill — APPROVED** [review]

**Pull request merged**